src/lib/operators/jit_operator/specialization/jit_code_specializer.hpp

```diff
@@ -0,0 +1,70 @@
+#pragma once
+
+#include <llvm/IR/Module.h>
+#include <llvm/IR/Verifier.h>
+
+#include <stack>
+
+#include "jit_compiler.hpp"
+#include "jit_repository.hpp"
+#include "jit_runtime_pointer.hpp"
+#include "llvm_extensions.hpp"
+
+namespace opossum {
+
+/* This is a helper class to retrieve the class name from a raw pointer using RTTI.
+ * The class has a single virtual method.
+ * A raw pointer can be cast to a pointer of this class to be used with RTTI.
+ */
+class JitRTTIHelper {
```

Comment:
What is the problem with typeid?

Reply:
I use typeid, but typeid already requires a typed pointer while I have a raw address that is computed by simulating bitcode instructions.
So I bet the pointer as an integer and then need to cast it to a pointer of some virtual class to make it usable with typeid.

Reply:
Can you store the type alongside with the pointer, similar to what happens with variants?

Reply:
This does not really work, since we need to relate LLVM types (LLVMs internal representation of the types in the bitcode) to "real" runtime types.
More fundamentally, at some point we want to marry LLVM bitcode with runtime data. This only works because both originate from the same original source code. But while we "know" this, there is no type-safe way to accomplish this, since both bitcode and runtime objects are unrelated from the compilers POV. So even if we get rid of `reinterpret_cast`s there will always be unsafe operations, I think.